docs/rules/require-expect.md

```diff
@@ -0,0 +1,84 @@
+# Ensure that `expect` is called when using assertions outside of top-level test callback
```

Comment:
I'm thinking we can simplify this description: "Ensure that `expect` is called" (the idea being "always" mode would be default anyway, and the option to change the behavior will appear later)